internals/overlord/stateengine.go

```diff
@@ -92,7 +129,7 @@ func (se *StateEngine) Ensure() error {
 	se.mgrLock.Lock()
 	defer se.mgrLock.Unlock()
 	if se.stopped {
```

Comment:
If you look at the ```snapd``` version of ```StartUp``` you will see they added a check inside ```Ensure``` to enforce running ```StartUp``` before ```Ensure```. Since our ```DryStart``` is very similar, in that it may initialise critical state before the manager may be able to run, should we do the same? Since ```DryStart``` is in ```New()``` while ```Ensure``` happens later, it seems like this will not be something likely to go wrong.